src/main/java/duke/utils/Parser.java

```diff
@@ -0,0 +1,177 @@
+package duke.utils;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import duke.Command;
+import duke.DukeException;
+import duke.DukeInvalidCommandException;
+import duke.DukeMissingArgumentException;
+import duke.DukeMissingTaskException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
```

Comment:
I liked that you separated the duke imports and the java imports, allows readers to understand the dependencies more easily.